pkg/sql/txn_restart_test.go

```diff
@@ -1708,27 +1708,76 @@ CREATE TABLE t.test (k INT PRIMARY KEY);
 				//   - a marker statement recognized by the filter.
 				//   - a write that will conflict with the read done by the filter and
 				//     will cause the txn's timestamp to be pushed.
-				if _, err := tx.Exec(
+				_, err = tx.Exec(
 					fmt.Sprintf(
 						`SHOW DATABASES; SELECT '%s'; INSERT INTO t.test VALUES (%d)`,
-						marker, i)); err != nil {
-					t.Fatal(err)
-				}
-
-				err = tx.Commit()
+						marker, i))
 				if moveOutOfAutoRetry {
 					if !isRetryableErr(err) {
 						t.Fatalf("expected retryable error, got: %v", err)
 					}
+					if err := tx.Rollback(); err != nil {
```

Comment:
My understanding is that this txn couldn't commit now, so I just Rollback it.

Sorry, I couldn't quite understand what do you mean here.

Reply:
This looks good. Never mind what I was saying.